libs/sdk-ui-charts/src/highcharts/chartTypes/warterfallChart/waterfallChartsSeries.ts

```diff
@@ -0,0 +1,101 @@
+    viewByAttribute: IUnwrappedAttributeHeadersWithItems,
+    colorStrategy: IColorStrategy,
+    type: string,
+    emptyHeaderTitle: string,
+): IPointData[] {
+    return seriesItem.map((pointValue: string, pointIndex: number) => {
+        // by default seriesIndex corresponds to measureGroup label index
+        let measureIndex = seriesIndex;
+
+        if (isOneOfTypes(type, multiMeasuresAlternatingTypes) && !viewByAttribute) {
+            measureIndex = pointIndex;
+        }
+
+        const pointData: IPointData = Object.assign(
+            {
+                y: parseValue(pointValue),
+                format: unwrap(measureGroup.items[measureIndex]).format,
+            },
+            pointValue === null
+                ? {
+                      marker: {
+                          enabled: false,
+                      },
+                  }
+                : {},
+        );
+
+        pointData.name = valueWithEmptyHandling(
+            unwrap(measureGroup.items[measureIndex]).name,
+            emptyHeaderTitle,
+        );
+        pointData.legendIndex = pointIndex;
+        pointData.color = colorStrategy.getColorByIndex(pointData.y < 0 ? 2 : 1);
+
+        return pointData;
+    });
+}
+
+export function getWaterfallChartSeries(
+    dv: DataViewFacade,
+    measureGroup: IMeasureGroupDescriptor["measureGroupHeader"],
+    viewByAttribute: IUnwrappedAttributeHeadersWithItems,
+    type: string,
+    colorStrategy: IColorStrategy,
+    emptyHeaderTitle: string,
+) {
+    return dv
+        .rawData()
+        .twoDimData()
+        .map((seriesItem: string[], seriesIndex: number) => {
+            const seriesItemData = getSeriesItemData(
+                seriesItem,
+                seriesIndex,
+                measureGroup,
+                viewByAttribute,
+                colorStrategy,
+                type,
+                emptyHeaderTitle,
+            );
+
+            const seriesItemConfig: ISeriesItemConfig = {
+                upColor: colorStrategy.getColorByIndex(1), //Positive color,
+                color: colorStrategy.getColorByIndex(2), //Negative color,
+                legendIndex: seriesIndex,
+                data: seriesItemData,
+                seriesIndex,
+            };
+
+            if (isOneOfTypes(type, multiMeasuresAlternatingTypes) && !viewByAttribute) {
+                seriesItemConfig.name = measureGroup.items
+                    .map((wrappedMeasure) =>
+                        valueWithEmptyHandling(unwrap(wrappedMeasure).name, emptyHeaderTitle),
+                    )
+                    .join(", ");
+            } else {
+                // otherwise seriesName is a measure name of index seriesIndex
+                seriesItemConfig.name = valueWithEmptyHandling(
+                    measureGroup.items[seriesIndex].measureHeaderItem.name,
+                    emptyHeaderTitle,
+                );
+            }
```

Comment:
I would create a new function that will return name value as string and call this method to get the value and put it above into construction of `seriesItemConfig` constant. It would look better if you create your object at once and not modify it later. It will be more readable.

Reply:
I updated this. Thank you!